### ISSUE TITLE
Feature Request: Time Selection in "flatpickr-date" Module for Date Range Filters, and Support for End Date/Time Field

### ISSUE THREAD

**Issue by serge402** [saltcorn/saltcorn#1903]
**Is your feature request related to a problem? Please describe.**
There is an issue with the "flatpickr-date" module when used as a Date Range Filter View in Saltcorn. The module lacks an option for time selection, limiting its functionality to date selection only. This poses a challenge when trying to filter by both date and time, particularly for scenarios such as filtering business open hours.
![Capture3](https://github.com/saltcorn/saltcorn/assets/66882139/7a8352a3-e992-447d-a807-afd014b1fb2e)

**Describe the solution you'd like**
I propose two enhancements to the "flatpickr-date" module in the Date Range Filter View:

1. Add the option for time selection within the "flatpickr-date" module when used as a Date Range Filter View. This would provide users with the ability to filter by both date and time, expanding the module's functionality and enabling more precise filtering.

2. Include support for an end_date/time field in the View filter settings. Currently, only the start_date/time field is available for filtering. By adding an end_date/time field, users would have the ability to filter for specific date/time points rather then ranges, offering greater flexibility in filtering options.

**Describe alternatives you've considered**
Not aware of alternatives within Saltcorn to address this limitation. Users may resort to custom coding or utilizing external libraries to achieve the desired functionality, which can be time-consuming and complex.

**Additional context**
By enhancing the "flatpickr-date" module in the Date Range Filter View, Saltcorn can provide users with a comprehensive solution for filtering by both date and time. This improvement would greatly benefit applications that require precise filtering based on date/time, such as managing business open hours or scheduling events.

**Comment by mark-myer** [comment]
I agree with @serge402. I would also appreciate some enhancements to the date range filter view. Specifically item number two, include support for an end_date/time field. However, in the project I'm doing, I do not want to filter by time of day. So I suggest having an option to turn time of day filtering on and off in the initial configuration. The same for the flatpickrdate module.

**Comment by serge402** [comment]
@mark-myer Have you tried the daterangepicker module? It supports the start_date and end_date fields.

But I don't use it as it has no time selection, which seems to be not an issue for you.

**Comment by mark-myer** [comment]
Maybe I don't quite understand. This is what the daterangefilter looks like now: 
<img width="912" alt="date-range-picker" src="https://github.com/saltcorn/saltcorn/assets/127981848/68355adb-d7f7-4544-a3a2-edfe8c2ec236">
I agree with you that time selection would be an enhancement, although it mostly gets in the way in the project I'm doing. However, I want to filter by long ranges; several years, usually. I propose an option like this, separating start and end dates (and times) into two different fields: 
<img width="914" alt="date-range-picker-separated" src="https://github.com/saltcorn/saltcorn/assets/127981848/27da9c3d-f4c9-4606-a76d-c69b3ccb1c2c">
Maybe this is already an option, and I have overlooked it. I tried to implement this with an extra state formula on a filter, but was not successful--there is precious little documentation about extra state formulas. I'd even be fine with two fields that the user could edit the date. They wouldn't have to be a date pickers, to suit me. Maybe I'm not understanding what you're trying to do. If so, I'm willing to start a new issue for this. 
Any help, @glutamate, or anyone else?

**Comment by serge402** [comment]
@mark-myer If I understand correctly, you should be able to create what you need there with current capabilities.

Create a Page with a List view in the middle, then add two separate Views in the top (Left and Right) like you have in the screenshot.
For the top left view, make it a Filter View on the start_date. For the top right view, make it a Filter View on the end_date.
That should allow you to double filter the list view in the middle.
Let me know if that helps.

Additionally, you can always use presents in the daterangefilter to setup pre-sets for 1, 2, 3 years from "Today" or for 2020, 2021 or 2020-2023, etc. Just edit how many days you want from the start of any particular year/month/week/day.
![image](https://github.com/saltcorn/saltcorn/assets/66882139/44cb840f-e873-44f2-b5da-a4274a67dee4)

**Comment by mark-myer** [comment]
Thanks for explaining that. My project took me in another direction, but I appreciate the help especially with the presets. If the flatpickr date module does get changed to include hours and minutes, I would be glad if there was an option to turn it on and off. 
Thanks!
